crmsh/migration.py

```diff
@@ -0,0 +1,655 @@
+        self.has_problems = True
+        self.block_migration = self.block_migration or is_blocker
+        match level:
+            case self.LEVEL_ERROR:
+                self.write_in_color(sys.stdout, constants.YELLOW, '[FAIL] ')
+            case self.LEVEL_WARN:
+                self.write_in_color(sys.stdout, constants.YELLOW, '[WARN] ')
+        print(title)
+        for line in details:
+            sys.stdout.write('       ')
+            print(line)
+
+    @staticmethod
+    def write_in_color(f, color: str, text: str):
+        if f.isatty():
+            f.write(color)
+            f.write(text)
+            f.write(constants.END)
+        else:
+            f.write(text)
+
+    def end(self):
+        sys.stdout.write('\n')
+
+
+def migrate():
+    try:
+        match _check_impl(local=False, json='', summary=False):
+            case 0:
+                logger.info("This cluster works on SLES 16. No migration is needed.")
+                return 0
+            case 1:
+                logger.info('Starting migration...')
+                migrate_corosync_conf()
+                logger.info('Finished migration.')
+                return 0
+            case _:
+                raise MigrationFailure('Unable to start migration.')
+    except MigrationFailure as e:
+        logger.error('%s', e)
+        return 1
+
+
+def check(args: typing.Sequence[str]) -> int:
+    parser = argparse.ArgumentParser(args[0])
+    parser.add_argument('--json', nargs='?', const='pretty', choices=['oneline', 'pretty'])
+    parser.add_argument('--local', action='store_true')
+    parsed_args = parser.parse_args(args[1:])
+    ret = _check_impl(parsed_args.local or parsed_args.json is not None, parsed_args.json, parsed_args.json is None)
+    if not parsed_args.json:
+        print('****** summary ******')
+        match ret:
+            case 0:
+                CheckResultInteractiveHandler.write_in_color(sys.stdout, constants.GREEN, '[INFO]')
+                sys.stdout.write(' This cluster works on SLES 16. No migration is needed.\n')
+            case 1:
+                CheckResultInteractiveHandler.write_in_color(sys.stdout, constants.GREEN, '[INFO]')
+                sys.stdout.write(' Please run "crm cluster health sles16 --fix" on on any one of above nodes.\n')
+                CheckResultInteractiveHandler.write_in_color(sys.stdout, constants.GREEN, '[PASS]')
+                sys.stdout.write(' This cluster is good to migrate to SLES 16.\n')
+            case _:
+                CheckResultInteractiveHandler.write_in_color(sys.stdout, constants.RED, '[FAIL]')
+                sys.stdout.write(' The pacemaker cluster stack can not migrate to SLES 16.\n')
+    return ret
+
+
+def _check_impl(local: bool, json: str, summary: bool) -> int:
+    """Return:
+    * 0 if the cluster is already on SLES 16.
+    * 1 if the cluster is good to migrate to SLES 16.
+    * 2 if the cluster is not good to migrate to SLES 16."""
```

Comment:
Better to create enum to make these return codes more describable, since other function like `migration` will also use these values